feat!: Rename and document common tokens for links (#1870)
Amsterdam/design-system: storybook/src/foundation/design-tokens/links.docs.mdx

{/* @license CC0-1.0 */}

import { Meta } from "@storybook/blocks";

<Meta title="Brand/Design Tokens/Links" />

# Links

> We have defined a group of link styles for all link components.

Reviewer: This mentions the reason (multiple link components) before the solution (common tokens), emphasizes the relation they codify, and works around using ‘we’.

```suggestion
Several link components share the same styles.
This set of common tokens represents that relation.
```

> This includes a common set of styles,
> along with specific styles for ‘regular’ and ‘subtle’ links.
> Links can also have a contrast colour for light (but not white) backgrounds
> or an inverse colour for dark backgrounds.
> 
> ## Default

Reviewer: I started removing the ‘regular’ here, but concluded that this doesn’t add much to the updated lead paragraph and the headings and introductions below. I think we can delete it.

```suggestion
## Default
```

Our default links have the following styles:

| CSS variable                                               | Value                                | Description                                                                       |
| :--------------------------------------------------------- | :----------------------------------- | :-------------------------------------------------------------------------------- |
| `var(--ams-links-color)`                                   | `var(--ams-color-interactive)`       | The default colour for links is our interactive colour.                           |
| `var(--ams-links-text-decoration-thickness)`               | `var(--ams-border-width-md)`         | Link underlines are our medium border width.                                      |

> | `var(--ams-links-text-underline-offset)`                   | `0.3333em`                           |                                                                                   |

Reviewer: Why not also mention that ‘The underline is positioned further below the text’.

| `var(--ams-links-hover-color)`                             | `var(--ams-color-interactive-hover)` | The hover colour is our interactive hover colour.                                 |
| `var(--ams-links-regular-hover-text-decoration-thickness)` | `var(--ams-border-width-lg)`         | The underline gets thicker on hover.                                              |

> | `var(--ams-links-regular-hover-text-underline-offset)`     | `0.2778em`                           | The offset decreases when you hover, making the underline look like it is rising. |

Reviewer: Two occurrences of ‘regular’ here. Can’t suggest because the table layout will change.

## Subtle

Subtle links are used in places where the context helps indicate that the text is a link.
For instance, in the Header or Pagination components.
As a result, their styling can be less obvious.

| CSS variable                                         | Value       | Description                                                                                                          |
| :--------------------------------------------------- | :---------- | :------------------------------------------------------------------------------------------------------------------- |
| `var(--ams-links-subtle-text-decoration-line)`       | `none`      | Subtle links do not show an underline by default. The surrounding context should clearly indicate that it is a link. |
| `var(--ams-links-subtle-hover-text-decoration-line)` | `underline` | An underline is shown on hover.                                                                                      |

## Contrast

> On light but not white backgrounds (like yellow or lime), the interactive contrast colour is used for links.

Reviewer: Shorter and more active, and removing the examples:

```suggestion
Links on light (but not white) backgrounds use the interactive contrast colour.
```

| CSS variable                            | Value                                   | Description                                                                                                   |
| :-------------------------------------- | :-------------------------------------- | :------------------------------------------------------------------------------------------------------------ |
| `var(--ams-links-contrast-color)`       | `var(--ams-color-interactive-contrast)` | On light backgrounds, the interactive contrast colour is used.                                                |
| `var(--ams-links-contrast-hover-color)` | `var(--ams-color-interactive-contrast)` | The same colour is used on hover. We currently do not show a difference between the default and hover states. |

## Inverse

> On dark backgrounds, the inverse colour is used for links.

Reviewer: ```suggestion
On dark backgrounds, links have the inverse colour.
```
